File: backend/TravelService/TravelService/Models/DbModels/Expense.cs
namespace TravelService.Models.DbModels
{
    public class Expense
    {
        public int Id { get; set; }
        public int TravelPlanId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Category { get; set; } = "OTHER";
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; } = string.Empty;

        public TravelPlan TravelPlan { get; set; } = null!;
    }
}

File: frontend/src/models/Expense.js
// models/Expense.js

/**
 * @typedef {Object} Expense
 * @property {number} id
 * @property {number} travelPlanId
 * @property {string} name
 * @property {'TRANSPORT'|'ACCOMMODATION'|'FOOD'|'TICKETS'|'SHOPPING'|'OTHER'} category
 * @property {number} amount
 * @property {string} date
 * @property {string} description
 */

/**
 * @typedef {Object} BudgetSummary
 * @property {number} plannedBudget
 * @property {number} totalSpent
 * @property {number} remainingBudget
 */